packages/block-library/src/playlist/index.php

```diff
@@ -0,0 +1,127 @@
+ */
+function render_block_core_playlist( $attributes, $content ) {
+	if ( empty( $attributes['tracks'] ) ) {
+		return '';
+	}
+
+	$current_id = $attributes['currentTrack'];
+
+	/**
+	 * Return early if no valid track ID is found.
+	 * This can happen if the user deleted all tracks but kept an empty inner block, such as the media upload placeholder.
+	 */
+	if ( empty( $current_id ) ) {
+		return '';
+	}
+
+	wp_enqueue_script_module( '@wordpress/block-library/playlist/view' );
+
+	$attachment_meta = wp_get_attachment_metadata( $current_id );
+	$current_title   = get_the_title( $current_id ) ? get_the_title( $current_id ) : '';
+	$current_artist  = isset( $attachment_meta['artist'] ) ? $attachment_meta['artist'] : '';
+	$current_album   = isset( $attachment_meta['album'] ) ? $attachment_meta['album'] : '';
+	$current_image   = isset( $attachment_meta['poster'] ) ? $attachment_meta['poster'] : '';
+	$current_url     = wp_get_attachment_url( $current_id );
+	$show_images     = isset( $attributes['showImages'] ) ? $attributes['showImages'] : false;
+	$aria_label      = $current_title;
+
+	if ( $current_title && $current_artist && $current_album ) {
+		$aria_label = sprintf(
+			/* translators: %1$s: track title, %2$s artist name, %3$s: album name. */
+			_x( '%1$s by %2$s from the album %3$s', 'track title, artist name, album name' ),
+			$current_title,
+			$current_artist,
+			$current_album
+		);
+	}
+
+	wp_interactivity_state(
+		'core/playlist',
+		array(
+			'currentID'     => $current_id,
+			'currentURL'    => $current_url,
+			'currentTitle'  => $current_title,
+			'currentAlbum'  => $current_album,
+			'currentArtist' => $current_artist,
+			'currentImage'  => $current_image,
+			'ariaLabel'     => $aria_label,
+		)
+	);
+
+	// Add the markup for the current track
+	$html = '<div><div class="wp-block-playlist__current-item">';
+	if ( $show_images && $current_image ) {
+		$html .=
+		'<img
+			class="wp-block-playlist__item-image"
+			src={ ' . esc_url( $current_image ) . '}
+			alt=""
+			width="70px"
+			height="70px"
+			data-wp-bind--src="state.currentImage"
+		/>';
+	}
+	$html .= '
+		<div>
+			<span class="wp-block-playlist__item-title" data-wp-text="state.currentTitle">' . $current_title . '</span>
+			<div class="wp-block-playlist__current-item-artist-album">
+				<span class="wp-block-playlist__item-artist" data-wp-text="state.currentArtist">' . $current_artist . '</span>
+				<span class="wp-block-playlist__item-album" data-wp-text="state.currentAlbum">' . $current_album . '</span>
+			</div>
+		</div>';
+	$html .= '</div><audio controls="controls" src="' . esc_url( $current_url ) . '"
+		data-wp-bind--src="state.currentURL"
+		data-wp-bind--aria-label="state.ariaLabel"
+		data-wp-watch="callbacks.init"
+		"/></audio></div>';
+
+	$figure = null;
+	preg_match( '/<figure[^>]*>/', $content, $figure );
+	if ( ! empty( $figure[0] ) ) {
+		$content = preg_replace( '/(<figure[^>]*>)/', '$1' . $html, $content, 1 );
+	}
```

Comment:
I am not trying to add a figure.
I am finding the figure to insert the HTML for the current track inside the figure.

There are two reasons:
- Because I was in the process of making the block dynamic, the save.js did not have the same access to the information, and did not include the markup for the current track, so I had to add it to the PHP file.

- Even before this change, when I was using the HTML API to add the wp-data attributes, the HTML API would add the attributes to the wrong HTML element. The title ended up as the album etc.
I could probably have solved this with some more time, but because of the mentioned change, I did not need to continue working on it at that point.